test/user.test.js

```diff
@@ -1606,6 +1606,42 @@ describe('User', function() {
           done();
         });
 
+        it('returns same verifyOptions after verify user model', function(done) {
+          const defaultOptions = {
+            type: 'email',
+            from: 'test@example.com',
+          };
+          var verifyOptions = Object.assign({}, defaultOptions);
+          const user = new User({
+            email: 'example@example.com',
+            password: 'pass',
+            verificationToken: 'example-token',
+          });
+          user
+            .verify(verifyOptions)
+            .then(function(res) {
+              expect(verifyOptions).to.eql(defaultOptions);
+              done();
```

Comment:
Mocha supports promises out of the box, there is no need to use `done` callback. Just return the promise from the test function.

```js
it('returns same verifyOptions after verify user model', () => {
  // ...
  return user
    .verify(verifyOptions)
    .then(res => expect(verifyOptions).to.eql(defaultOptions));
});
```

Also note that we prefer arrow functions in new code, see http://loopback.io/doc/en/contrib/style-guide.html#arrow-functions

I'll fix this myself to speed things up.